controllers/api/file/PKPManageFileApiHandler.php

```diff
@@ -88,6 +90,51 @@ public function deleteFile($args, $request)
         return \PKP\db\DAO::getDataChangedEvent();
     }
 
+	/**
+	 * Restore original file when cancelling the upload wizard
+	 */
+	public function cancelFileUpload(array $args, Request $request): JSONMessage
+	{
+		if (!$request->checkCSRF()) {
+			return new JSONMessage(false);
+		}
+
+		$submissionFile = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION_FILE);
+		$originalFile = $request->getUserVar('originalFile') ? (array) $request->getUserVar('originalFile') : null;
+		$revisedFileId = $request->getUserVar('fileId') ? (int) $request->getUserVar('fileId') : null;
+
+		// Get revisions and check file IDs
+		$revisions = Repo::submissionFile()->getRevisions($submissionFile->getId());
+		$revisionIds = [];
+		foreach ($revisions as $revision) {
+			$revisionIds[] = $revision->fileId;
+		}
+
+		if (!$revisedFileId && !in_array($revisedFileId, $revisionIds)) {
+			return new JSONMessage(false);
+		}
+
+		if (!isset($originalFile['fileId']) && !in_array($originalFile['fileId'], $revisionIds)) {
+			return new JSONMessage(false);
+		}
+
+		// Restore original submission file
+		Repo::submissionFile()->edit(
+			$submissionFile,
+			[
+				'fileId' => $originalFile['fileId'],
```

Comment:
This will work. But in theory I could use this to switch to any revision, not just the last revision. It wouldn't cause any big problems. But instead of getting the revision from a request user var, you could reset the `fileId` to the most recent of the `$revisionIds`.

The method `getRevisions()` returns with the most recent first, so you could use the first revision id in that array.

Reply:
Yes, I thought about that.
User is able to upload indefinite number of revisions through the submission file wizard, thus, I don't know which revision I need to retrieve.